src/typedefs.rs

```diff
@@ -193,6 +193,18 @@ macro_rules! func_len (() => (
     }
 ));
 
+/// Macro to implement a `is_empty()` function which will return `true` if `self.len() == 0`.
```

Comment:
Small note: maybe "an" here instead of "a" before `is_empty`.

Reply:
Nice spot! Fixed this.